lib-multisrc/clipstudioreader/src/eu/kanade/tachiyomi/multisrc/clipstudioreader/ClipStudioReader.kt

```diff
@@ -0,0 +1,189 @@
+package eu.kanade.tachiyomi.multisrc.clipstudioreader
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.parseAs
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+
+abstract class ClipStudioReader(
+    override val name: String,
+    override val baseUrl: String,
+    override val lang: String,
+) : HttpSource() {
+    override val supportsLatest = false
+
+    override val client = super.client.newBuilder()
+        .addInterceptor(Deobfuscator())
+        .addInterceptor(ImageInterceptor())
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .set("Referer", "$baseUrl/")
+
+    override fun pageListParse(response: Response): List<Page> {
+        val requestUrl = response.request.url
+        val contentId = requestUrl.queryParameter("c")
+
+        if (contentId != null) {
+            // EPUB-based path
+            val tokenUrl = "$baseUrl/api/tokens/viewer?content_id=$contentId".toHttpUrl()
+            val tokenResponse = client.newCall(GET(tokenUrl, headers)).execute()
+            val viewerToken = tokenResponse.parseAs<TokenResponse>().token
+
+            val metaUrl = "$baseUrl/api/contents/$contentId/meta".toHttpUrl()
+            val apiHeaders = headersBuilder().add("Authorization", "Bearer $viewerToken").build()
+            val metaResponse = client.newCall(GET(metaUrl, apiHeaders)).execute()
+            val contentBaseUrl = metaResponse.parseAs<MetaResponse>().content.baseUrl
+
+            val preprocessUrl = "$contentBaseUrl/preprocess-settings.json"
+            val obfuscationResponse = client.newCall(GET(preprocessUrl, headers)).execute()
+            val obfuscationKey = obfuscationResponse.parseAs<PreprocessSettings>().obfuscateImageKey
+
+            val containerUrl = "$contentBaseUrl/META-INF/container.xml"
+            val containerResponse = client.newCall(GET(containerUrl, headers)).execute()
+            val containerDoc = containerResponse.asJsoup()
```

Comment:
```suggestion
            val containerDoc = Jsoup.parse(containerResponse.body.string(), containerUrl, Parser.xmlParser())
```
use xml parser for xml, may need to adjust the selector